File: php/muse/algorithms/number_theory/Parity.php
<?php

declare(strict_types=1);

namespace muse\algorithms\number_theory\Parity;

function moduloIsEven(int $n): bool
{
    return $n % 2 == 0;
}

function moduloIsOdd(int $n): bool
{
    return $n % 2 != 0;
}

function bitwiseIsEven(int $n): bool
{
    return ($n & 1) == 0;
}

function bitwiseIsOdd(int $n): bool
{
    return ($n & 1) != 0;
}

File: archive/ruby/muse/algorithms/number_theory/Parity.rb
module Parity
  def self.moduloIsEven(n)
    (n % 2).zero?
  end

  def self.moduloIsOdd(n)
    n.odd?
  end

  def self.bitwiseIsEven(n)
    (n & 1).zero?
  end

  def self.bitwiseIsOdd(n)
    (n & 1) != 0
  end
end
